src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1406,25 +1373,23 @@ enum SubtypingRelationKind {
             REF_VAL_ALLOWED
         }
 
-        public abstract static class ParameterizedTypeRelation<P> extends TypeRelation {
-            P param;
+        SubtypingRelation isSubtypeRelation = new SubtypingRelation();
+        class SubtypingRelation extends TypeRelation {
+            SubtypingRelationKind param;
```

Comment:
What is this needed for? We should get into a position where subtyping is "just" subtyping, no need for having some kind of mode flag - unless I'm misunderstanding something?

Reply:
yep I will tackle that now, if I recall correctly I tried to removed it but there where other changes in areas a bit distant from the visitors that's why I didn't go for it in a first step